### DISCUSSION TITLE
Coach Models and RVS devices exposed

### DISCUSSION THREAD

**Discussion by rlust** [spbrogan/rvc2mqtt#110] [Q&A] [answered]
Would be interesting to see what coach you are using to develop or test this integration.
What devices are exposed to the RVC CAN bus?

**Comment by rlust** [comment]
I have a 2022 Entegra Aspire.
Devices exposed on RVC bus:
Awnings
Lights
Generator
Water Heater 
Inverter
Tank Status
DC Charger
DC Battery
Air Conditioner Status

**Reply by spbrogan** [reply]
for supported devices you can see here.
https://github.com/spbrogan/rvc2mqtt/tree/main/rvc2mqtt/entity

**Reply by spbrogan** [reply]
Adding new devices is pretty easy depending on how complex the device is and how much you want to expose to HA.  I have generally added all the data in HA only to find my RVC devices only use a very small subset.  Would be really nice if the RV industry wasn't such a mess.

**Comment by rlust** [comment]
I am also getting RVC instance states from devices that are unknown to my current integration using a eRVIN pi system.

Here is a list of instances that are unknown:

UNKNOWN-1FF5A = {"data":"998809047EF31D00","dgn":"1FF5A","name":"UNKNOWN-1FF5A","timestamp":"1645977377.945780"}
UNKNOWN-1FF58 = {"data":"190000007DFFFF32","dgn":"1FF58","name":"UNKNOWN-1FF58","timestamp":"1645977377.485584"}
UNKNOWN-1FACE = {"data":"02FF00FFFFFFFFFA","dgn":"1FACE","name":"UNKNOWN-1FACE","timestamp":"1645977377.561858"}
UNKNOWN-17EFE = {"data":"303A6E755278614D","dgn":"17EFE","name":"UNKNOWN-17EFE","timestamp":"1645977376.850988"}
UNKNOWN-0FF2E = {"data":"80052200FFFF0000","dgn":"0FF2E","name":"UNKNOWN-0FF2E","timestamp":"1645977377.716669"}
UNKNOWN-1FF56 = {"data":"1908AC06007D0000","dgn":"1FF56","name":"UNKNOWN-1FF56","timestamp":"1645977377.600066"}
UNKNOWN-15FCE = {"data":"B600000000000000","dgn":"15FCE","name":"UNKNOWN-15FCE","timestamp":"1645977377.359279"}
UNKNOWN-1FF0A = {"data":"19000004FFFFFFFF","dgn":"1FF0A","name":"UNKNOWN-1FF0A","timestamp":"1645977377.714679"}
UNKNOWN-1FACF = {"data":"49440C080306FF05","dgn":"1FACF","name":"UNKNOWN-1FACF","timestamp":"1645977373.374159"}
UNKNOWN-1AAFD = {"data":"02FF120100000000","dgn":"1AAFD","name":"UNKNOWN-1AAFD","timestamp":"1645977377.358703"}
UNKNOWN-1FE98 = {"data":"011FFFFFFFFFFFFF","dgn":"1FE98","name":"UNKNOWN-1FE98","timestamp":"1645977377.171646"}
UNKNOWN-0E8FF = {"data":"00FFFFFFFFDBFE01","dgn":"0E8FF","name":"UNKNOWN-0E8FF","timestamp":"1645977377.379667"}
UNKNOWN-1FBDA = {"data":"A901FFFFFFFFFFFF","dgn":"1FBDA","name":"UNKNOWN-1FBDA","timestamp":"1645977377.379089"}
UNKNOWN-1FE99 = {"data":"0110000000000014","dgn":"1FE99","name":"UNKNOWN-1FE99","timestamp":"1645977377.199241"}
UNKNOWN-1FE97 = {"data":"0100000000001000","dgn":"1FE97","name":"UNKNOWN-1FE97","timestamp":"1645977377.199820"}
UNKNOWN-0FEF5 = {"data":"FFFFFF4025FFFFFF","dgn":"0FEF5","name":"UNKNOWN-0FEF5","timestamp":"1645977377.992261"}
UNKNOWN-0FEFC = {"data":"FF98FFFFFFFFFFFF","dgn":"0FEFC","name":"UNKNOWN-0FEFC","timestamp":"1645977377.092208"}
UNKNOWN-1FACD = {"data":"FFFFFFFF03FFFFFA","dgn":"1FACD","name":"UNKNOWN-1FACD","timestamp":"1645977375.383061"}
UNKNOWN-14390 = {"data":"2124410000360000","dgn":"14390","name":"UNKNOWN-14390","timestamp":"1645977377.825413"}
UNKNOWN-0EAFF

**Reply by spbrogan** [reply, marked as answer]
I have updated this projects version of RVC spec and the logic to decode a few additional DGNs (ack, terminal need different logic).  Below are the ones I can decode.  Other values i don't see in spec. 

17E - Is a terminal
1FF56 - unknown.  All dgn in the 1ff5 range is undefined in the spec i have
1FE98 - WATERHEATER_COMMAND_2 
0E8FF - Ack
1FE99 - WATERHEATER_STATUS_2
1FE97- CIRCULATION_PUMP_STATUS

**Comment by spbrogan** [comment]
> Would be interesting to see what coach you are using to develop or test this integration. What devices are exposed to the RVC CAN bus?

I have been using two different 2021 Keystone Cougar trailers.  The RVC layout is similar but not the same.  You can look here to get an idea of what is integrated so far.  https://github.com/spbrogan/rvc2mqtt/blob/main/docs/configuration.md#example

**Reply by rlust** [reply]
I am using a 2022 Entegra Aspire
